src/toxcore/dht/kbucket.rs

```diff
@@ -401,6 +401,15 @@ impl Kbucket {
         )
     }
 
+    /// return peer which has pk
+    pub fn find_node(&self, pk: &PublicKey) -> Option<DhtNode> {
+        self.bucket_index(pk).and_then(|index|
+            self.buckets[index]
+                .find(&self.pk, pk)
+                .map(|node_index| self.buckets[index].nodes[node_index].clone())
```

Comment:
I's better to return reference here. Then you can clone after calling this function if you really need it.

Reply:
done